### ISSUE TITLE
Usability: Segments that can't have children raise NoMethodError 

### ISSUE THREAD

**Issue by untoldone** [ruby-hl7/ruby-hl7#39]
While segments with children respond to `:children`, segments (such as MSH) do not. This seemed to not match my abstraction usability expectation. With the current implementation, I'll have to write code to detect if Segment has a child or not rather than just getting no children.

I do see that there is a :has_children? method which helps, but it just feels off to have to call a method like this to prevent this type of StandardError.

**Comment by mogox** [comment]
I agree this is an issue, I've seen it more than a few times, we probably want to fix this in a major release.
I think the solution is always have ```children``` and treat it as an empty collection for segments which do not have one.  I believe this would be change that will break compatibility for those who depend on the method ```children``` being only available for segments with children.

What do you think @Stratus3D ?

**Comment by untoldone** [comment]
One thought: You could always add another method in addition to `children` to avoid the reverse compat issue and depreciate `children`.

**Comment by mogox** [comment]
That's an excellent idea @untoldone.
